### ISSUE TITLE
new-lib: FileAdapter and SSL issue

### ISSUE THREAD

**Issue by zimbatm** [pusher/pusher-http-php#87]
`make test` returns the following error on my system:

```
1) Pusher\Tests\FileAdapterTest::testRequest
Notice: Undefined index: _SERVER in /Users/zimbatm/code/github.com/pusher/pusher-http-php/vendor/phpunit/phpunit/src/Util/Filter.php on line 28
Pusher\Exception\AdapterException: file_get_contents(https://pusher.com): failed to open stream: operation failed

/Users/zimbatm/code/github.com/pusher/pusher-http-php/src/Http/FileAdapter.php:73
/Users/zimbatm/code/github.com/pusher/pusher-http-php/tests/FileAdapterTest.php:38
/Users/zimbatm/code/github.com/pusher/pusher-http-php/vendor/phpunit/phpunit/phpunit:36
```

However SSL seem to be enabled and stream_get_contents() to support including remote files:

```
$ php -a
Interactive shell

php > var_dump(stream_get_wrappers());
array(12) {
  [0]=>
  string(5) "https"
  [1]=>
  string(4) "ftps"
  [2]=>
  string(13) "compress.zlib"
  [3]=>
  string(14) "compress.bzip2"
  [4]=>
  string(3) "php"
  [5]=>
  string(4) "file"
  [6]=>
  string(4) "glob"
  [7]=>
  string(4) "data"
  [8]=>
  string(4) "http"
  [9]=>
  string(3) "ftp"
  [10]=>
  string(4) "phar"
  [11]=>
  string(3) "zip"
}
php > var_dump(ini_get('allow_url_fopen'));
string(1) "1"
php > var_dump(extension_loaded('openssl'));
bool(true)
```

If I change the test domain to be google.com or github.com I get the same error.
Changing to plain HTTP doesn't have the same issue.

**Comment by GrahamCampbell** [comment]
What php version?

**Comment by GrahamCampbell** [comment]
Having the `cacert.pem` file in the code base is a bad practice btw.

**Comment by zimbatm** [comment]
```
$ php -v
PHP 5.5.30 (cli) (built: Oct 23 2015 17:21:45)
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2015 Zend Technologies
```

What is the best practice in regards to `cacert.pem` ? I know that PHP 5.6 is able to find the system CA certs but what to do with older versions ? I've been looking at Guzzle quite a lot and here is what they do: https://github.com/guzzle/guzzle/blob/master/src/Handler/StreamHandler.php#L352-L353

**Comment by zimbatm** [comment]
Btw if you have any ideas on how to improve the new-lib I'm all ears, I don't qualify as a PHP veteran at all.

**Comment by GrahamCampbell** [comment]
TBH, I'd actually recommend just using guzzle 5/6 and dropping php 5.3 support, but keeping php 5.4 support. Then you don't need to worry about any of this http stuff. Guzzle will do it all for us.

**Comment by GrahamCampbell** [comment]
NB, requiring guzzle `^5.0|^6.0` will allow either version to be resolved for maximum compatibility.

**Comment by GrahamCampbell** [comment]
Also note that guzzle 3 can be installed along side guzzle 4+ because it's technically a different package in a different namesace, so conflicts with older libraries shouldn't be a problem either.

**Comment by zimbatm** [comment]
It's tempting to use guzzle but it also introduces a lot of unnecessary complexity. My hope was to keep this library small and not add too much dependencies.

**Comment by GrahamCampbell** [comment]
Guzzle is small imo, and definitely makes this library smaller and simpler.

**Comment by zimbatm** [comment]
```
$ php -v
PHP 5.5.30 (cli) (built: Oct 23 2015 17:21:45)
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2015 Zend Technologies
```
What is the best practice in regards to `cacert.pem` ? I know that PHP 5.6 is able to find the system-one but what to do with older versions ?

**Comment by kn100** [comment]
Closed in preference to #95
